Python/opcode_metadata.h

```diff
@@ -0,0 +1,184 @@
+    [LOAD_FAST] = { 0, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LOAD_CONST] = { 0, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [STORE_FAST] = { 1, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LOAD_FAST__LOAD_FAST] = { 0, 2, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LOAD_FAST__LOAD_CONST] = { 0, 2, DIR_NONE, DIR_NONE, DIR_NONE },
+    [STORE_FAST__LOAD_FAST] = { 1, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [STORE_FAST__STORE_FAST] = { 2, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LOAD_CONST__LOAD_FAST] = { 0, 2, DIR_NONE, DIR_NONE, DIR_NONE },
+    [POP_TOP] = { 1, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [PUSH_NULL] = { 0, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [UNARY_POSITIVE] = { 1, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [UNARY_NEGATIVE] = { 1, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [UNARY_NOT] = { 1, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [UNARY_INVERT] = { 1, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP_MULTIPLY_INT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP_MULTIPLY_FLOAT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP_SUBTRACT_INT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP_SUBTRACT_FLOAT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP_ADD_UNICODE] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP_INPLACE_ADD_UNICODE] = { 2, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP_ADD_FLOAT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP_ADD_INT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_SUBSCR] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_SLICE] = { 3, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [STORE_SLICE] = { 4, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_SUBSCR_LIST_INT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_SUBSCR_TUPLE_INT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_SUBSCR_DICT] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_SUBSCR_GETITEM] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LIST_APPEND] = { 1, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [SET_ADD] = { 1, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [STORE_SUBSCR] = { 3, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [STORE_SUBSCR_LIST_INT] = { 3, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [STORE_SUBSCR_DICT] = { 3, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [DELETE_SUBSCR] = { 2, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [PRINT_EXPR] = { 1, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [RAISE_VARARGS] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [INTERPRETER_EXIT] = { 1, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [RETURN_VALUE] = { 1, 0, DIR_NONE, DIR_NONE, DIR_NONE },
+    [GET_AITER] = { 1, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [GET_ANEXT] = { 1, 2, DIR_NONE, DIR_NONE, DIR_NONE },
+    [GET_AWAITABLE] = { 1, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [SEND] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [ASYNC_GEN_WRAP] = { 1, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [YIELD_VALUE] = { 1, 0, DIR_NONE, DIR_NONE, DIR_NONE },
```

Comment:
[stack_effect of YIELD_VALUE](https://github.com/python/cpython/blob/15aecf8dd70f82eb507d74fae9662072a377bdc8/Python/compile.c#L1128) is 0. Why is it here 1 popped, 0 pushed?

Reply:
That seems to be a bug in bytecodes.c. The code ends with `goto resume_frame` so the push sequence at the end is never executed. I can fix it by making the IO effect `(retval -- retval)` (indicating that it doesn't change) or perhaps `(retval -- unused)`, indicating that we don't write it here. I presume it's written in some weird way when the frame is resumed.